sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -913,27 +920,18 @@ int getRetryTimeoutInMs(int retry) {
     }
 
     private Mono<Boolean> checkIMDSAvailable(String endpoint) {
-        return Mono.fromCallable(() -> {
-            HttpURLConnection connection = null;
-            URL url = getUrl(endpoint + "?api-version=2018-02-01");
-
-            try {
-                connection = (HttpURLConnection) url.openConnection();
-                connection.setRequestMethod("GET");
-                connection.setConnectTimeout(1000);
-                connection.connect();
-            } catch (Exception e) {
-                throw LoggingUtil.logCredentialUnavailableException(LOGGER, options,
-                    new CredentialUnavailableException("ManagedIdentityCredential authentication unavailable. "
-                        + "Connection to IMDS endpoint cannot be established, " + e.getMessage() + ".", e));
-            } finally {
-                if (connection != null) {
-                    connection.disconnect();
-                }
-            }
+        URL url = null;
+        try {
+            url = getUrl(endpoint + "?api-version=2018-02-01");
+        } catch (MalformedURLException e) {
+            return Mono.error(e);
+        }
+        HttpRequest request = new HttpRequest(HttpMethod.GET, url);
 
-            return true;
-        });
+        return getPipeline().send(request)
```

Comment:
This is used for probing, I believe.
We need to have a 1 second timeout on this request.